pkg/agent/agent.go

```diff
@@ -85,7 +86,7 @@ func CreateRelease(actionConfig *action.Configuration, name, namespace, valueStr
 	if err != nil {
 		// Simulate the Atomic flag and delete the release if failed
 		errDelete := DeleteRelease(actionConfig, name, false)
-		if errDelete != nil {
+		if errDelete != nil && !strings.Contains(errDelete.Error(), "release: not found") {
```

Comment:
Could we not do
```
import "helm.sh/helm/v3/pkg/storage/driver"
...
if errDelete != nil && err != driver.ErrReleaseNotFound {
...
```
rather than using string comparisons?

Reply:
We can't. The errro `ErrReleaseNotFound` is just `release: not found` but the error that `DeleteRelease` is wrapped with more details: `uninstall: Release not loaded: neat-shoe: release: not found`. If we compare both errors, they are not equal.